micropub.php

```diff
@@ -49,389 +51,419 @@
+					$args['post_category'] = array_merge(
+						$args['post_category'] ?: array(),
+						$add_args['post_category']
+					);
 				}
-			} elseif ( is_array( $delete ) ) {
-				foreach ( static::mp_to_wp( array( 'properties' => array_flip( $delete ) ) )
-						  as $name => $_ ) {
-					$args[ $name ] = NULL;
+			}
+
+			// replace
+			$replace = static::$input['replace'];
+			if ( $replace ) {
+				if ( ! is_array( $replace ) ) {
+					static::error( 400, 'replace must be an object' );
 				}
-				if ( in_array( 'category', $delete ) ) {
-					wp_set_post_tags( $post_id, '', false);
-					wp_set_post_categories( $post_id, '');
+				foreach ( static::mp_to_wp( array( 'properties' => $replace ) )
+					as $name => $val ) {
+					$args[ $name ] = $val;
 				}
-			} else {
-				static::error( 400, 'delete must be an array or object' );
 			}
-		}
 
-		// tell wordpress to preserve published date explicitly, otherwise
-		// wp_update_post sets it to the current time
-		$args['edit_date'] = true;
-		$args = static::store_mf2( static::store_geodata(
-			static::generate_post_content( $args ) ) );
-		if ( WP_DEBUG ) {
-			error_log( 'wp_update_post with args: ' . serialize( $args ) );
-		}
+			// delete
+			$delete = static::$input['delete'];
+			if ( $delete ) {
+				if ( is_assoc_array( $delete ) ) {
+					if ( array_diff( array_keys( $delete ), array( 'category', 'syndication' ) ) ) {
+						static::error( 400, 'can only delete individual values from category and syndication; other properties not supported' );
+					}
+					$delete_args = static::mp_to_wp( array( 'properties' => $delete ) );
+					if ( $delete_args['tags_input'] ) {
+						$args['tags_input'] = array_diff(
+							$args['tags_input'] ?: array(),
+							$delete_args['tags_input']
+						);
+					}
+					if ( $delete_args['post_category'] ) {
+						$args['post_category'] = array_diff(
+							$args['post_category'] ?: array(),
+							$delete_args['post_category']
+						);
+					}
+				} elseif ( is_array( $delete ) ) {
+					foreach ( static::mp_to_wp( array( 'properties' => array_flip( $delete ) ) )
+						as $name => $_ ) {
+						$args[ $name ] = null;
+					}
+					if ( in_array( 'category', $delete, true ) ) {
+						wp_set_post_tags( $post_id, '', false );
+						wp_set_post_categories( $post_id, '' );
+					}
+				} else {
+					static::error( 400, 'delete must be an array or object' );
+				}
+			}
 
-		kses_remove_filters();
-		static::check_error( wp_update_post( $args, true ) );
-		kses_init_filters();
+			// tell WordPress to preserve published date explicitly, otherwise
+			// wp_update_post sets it to the current time
+			$args['edit_date'] = true;
+			$args              = static::store_mf2(
+				static::store_geodata(
+					static::generate_post_content( $args )
+				)
+			);
+			if ( WP_DEBUG ) {
+				error_log( 'wp_update_post with args: ' . serialize( $args ) );
+			}
 
-		static::default_file_handler( $post_id );
```

Comment:
this one too. do you maybe need to rebase on master? it'll be easier to tell one way or the other after the global indent is reverted.